libcore/src/SimulationHelper.cpp

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2020 Forschungszentrum Jülich GmbH. All rights reserved.
+ *
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ **/
+#include "SimulationHelper.h"
+
+#include "general/Logger.h"
+#include "geometry/Room.h"
+#include "geometry/SubRoom.h"
+
+#include <algorithm>
+
+std::optional<bool> SimulationHelper::UpdateRoom(const Building & building, Pedestrian & ped)
```

Comment:
Maybe rename the function. At the moment it sounds like the room is updated, but the pedestrians room is updated.

Reply:
Changed the name